internal/controllers/clusterclass/clusterclass_controller.go

```diff
@@ -263,3 +326,52 @@ func (r *Reconciler) reconcileExternal(ctx context.Context, clusterClass *cluste
 func uniqueObjectRefKey(ref *corev1.ObjectReference) string {
 	return fmt.Sprintf("Name:%s, Namespace:%s, Kind:%s, APIVersion:%s", ref.Name, ref.Namespace, ref.Kind, ref.APIVersion)
 }
+
+// extensionConfigToClusterClass maps an ExtensionConfigs with the corresponding ClusterClass to reconcile them on updates
+// of the ExtensionConfig.
+func (r *Reconciler) extensionConfigToClusterClass(o client.Object) []reconcile.Request {
+	res := []ctrl.Request{}
+
+	ext, ok := o.(*runtimev1.ExtensionConfig)
+	if !ok {
+		panic(fmt.Sprintf("Expected an ExtensionConfig but got a %T", o))
+	}
+
+	clusterClasses := clusterv1.ClusterClassList{}
+	selector, err := metav1.LabelSelectorAsSelector(ext.Spec.NamespaceSelector)
+	if err != nil {
+		return nil
+	}
+	if err := r.Client.List(context.TODO(), &clusterClasses); err != nil {
+		return nil
+	}
+	for _, clusterClass := range clusterClasses.Items {
+		for _, patch := range clusterClass.Spec.Patches {
+			if patch.External != nil && patch.External.DiscoverVariablesExtension != nil {
```

Comment:
It would be good if we can not only check if the ClusterClass has DiscoverVariablesExtension set, but also that it references the current ExtensionConfig ext

Reply:
Let me take this as a follow-up. The conventions and validation around the names of these extensions isn't clear enough to rely on strings.split IMO, so it would be good to do that together.

Reply:
Good point!